src/Concordium/Client/GRPC2.hs

```diff
@@ -3480,19 +3481,19 @@ withGRPCCore helper k = do
                                 let runRPC =
                                         runExceptT (helper client')
                                             >>= \case
-                                                Left err -> Nothing <$ logm ("Network error: " <> fromString (show err)) -- client error
-                                                Right (Left err) -> Nothing <$ logm ("Too much concurrency: " <> fromString (show err))
-                                                Right (Right x) -> return (Just x)
+                                                Left err -> Left Retry <$ logm ("Network error: " <> fromString (show err)) -- client error
+                                                Right (Left err) -> Left (DoNotRetry (StatusNotOk (RESOURCE_EXHAUSTED, "Too many concurrent requests."))) <$ logm ("Too much concurrency: " <> fromString (show err))
```

Comment:
... Try again later?

Reply:
That's up to the caller of this function. We won't be doing that here.